akka-docs/rst/additional/osgi.rst

```diff
@@ -1,12 +1,100 @@
 Akka in OSGi
 ============
 
+Background
+----------
+
+`OSGi <www.osgi.org/developer>`_ is a mature packaging and deployment standard for component-based systems. It
+has similar capabilities as Project Jigsaw (originally scheduled for JDK 1.8), but has far stronger facilities to
+support legacy Java code. This is to say that while Jigsaw-ready modules require significant changes to most source files
+and on occasion to the structure of the overall application, OSGi can be used to modularize almost any Java code as far
+back as JDK 1.2, usually with no changes at all to the binaries.
+
+These legacy capabilities are OSGi's major strength and it's major weakness. The creators of OSGi realized early on that
+implementors would be unlikely to rush to support OSGi metadata in existing JARs. There were already a handful of new
+concepts to learn in the JRE and the added value to teams that were managing well with straight J2EE was not obvious.
+Facilities emerged to "wrap" binary JARs so they could be used as bundles, but this functionality was only used in limited
+situations. An application of the "80/20 Rule" here would ahve that "80% of the complexity is with 20% of the configuration",
+but it was enough to give OSGi a reputation that has stuck with it to this day.
+
+Today, the situation is much different. Most popular artifacts available in open source repositories are instrumented
+with OSGi bundle metadata and are tested and deployed in production environments. Generating the metadata is a reduced
+to a simple build step with permissive defaults via plugins that are available for all popular build tools. Various
+deployment options exist for embedding OSGi in the smallest Java applications (such as for plugin architectures) all
+the way up to enterprise containers with modular (of course!) command lines and comprehensive management interfaces.
```

Comment:
I am not sure this and the above paragraph belongs to Akka documentation. It is too much opinion in there that is outside of the scope of Akka IMHO.